### DISCUSSION TITLE
How to download a specified version of mise?

### DISCUSSION THREAD

**Discussion by rakuyoMo** [jdx/mise#1946] [Q&A]
Does mise have a configuration file similar to `.mise-version` for fixed versions?

Or does `.mise.toml` have any fields used to declare the version of mise?

**Comment by kattouf** [comment]
You can specify `MISE_VERSION` env variable when use `mise.run` [install script](https://mise.jdx.dev/getting-started.html#alternate-installation-methods)

**Comment by Marukome0743** [comment]
Following up with the current options: mise does not use a `.mise-version` file to pin the executable that is already running.

For most projects, the recommended approach is to declare the minimum supported mise version in `mise.toml`:

```toml
min_version = "2026.7.0"
```

This produces an error when the installed mise is older. A soft recommendation is also available:

```toml
min_version = { soft = "2026.7.0" }
```

If the project must bootstrap one exact mise release, the install script supports `MISE_VERSION`:

```sh
curl https://mise.run | MISE_VERSION=v2026.7.0 sh
```

Alternatively, mise can generate a project bootstrap script with the version embedded:

```sh
mise generate bootstrap --version v2026.7.0 --write ./bin/mise
```

The generated wrapper downloads and executes that version, which avoids the bootstrapping problem of requiring an existing mise binary to read its own version constraint.

In general, `min_version` is preferable when the project only requires certain mise features, while an exact bootstrap version is useful when fully reproducible onboarding is required.

I verified these options against the current main CLI and documentation.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
